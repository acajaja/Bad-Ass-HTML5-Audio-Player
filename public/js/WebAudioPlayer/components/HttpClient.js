export const get = async (endpoint, queryParams = null) => {
    let URL = endpoint;

    if (queryParams) {
        const qp = new URLSearchParams(queryParams);
        URL = `${URL}?${qp.toString}`;
    }

    const requestParams = {
        headers: new Headers({
            'Content-Type': 'application/json',
            'X-Requested-With': ''
        }),
        method: 'GET',
        mode: 'same-origin'
    };

    const response = await fetch(URL, requestParams);

    if (!response.ok) {
        throw new Error('Bad');
    }

    return await response.json();
}
