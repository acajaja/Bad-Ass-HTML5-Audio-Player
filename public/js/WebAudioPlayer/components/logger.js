/* istanbul ignore file */
const _log = (level, msg) => {
    console.log(msg);
}

export const debug = (msg) => {
    _log(1, msg);
}
export const log = (msg) => {
    _log(2, msg);
}
export const error = (msg) => {
    _log(3, msg);
}
export const info = (msg) => {
    _log(4, msg);
}
export const warn = (msg) => {
    _log(5, msg);
}
export const critical = (msg) => {
    _log(6, msg);
}
