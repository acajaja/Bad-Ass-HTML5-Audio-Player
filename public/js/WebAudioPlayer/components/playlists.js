/* istanbul ignore file */
import { PLAYLIST_BASE_URL } from './vars.js'
export const playlists = [
	{
		"name":"East 3rd Street Ensemble Samples",
		"url":`${PLAYLIST_BASE_URL}/playlists/east-third-st.json`
	},
	{
		"name":"Prajna Samples",
		"url":`${PLAYLIST_BASE_URL}/playlists/prajna.json`
	},
    {
		"name":"Zuk - An Evening in the Clouds",
		"url":`${PLAYLIST_BASE_URL}/playlists/zuk.json`
	},
    {
		"name":"1975: An Underground Epic Samples",
		"url":`${PLAYLIST_BASE_URL}/playlists/1975.json`
	},
    {
		"name":"Round Robin Sample",
		"url":`${PLAYLIST_BASE_URL}/playlists/round-robin.json`
	},
    {
		"name":"Marim-bass Samples",
		"url":`${PLAYLIST_BASE_URL}/playlists/marim-bass.json`
	},
    {
		"name":"Acoustic Mayhem Sample",
		"url":`${PLAYLIST_BASE_URL}/playlists/acoustic-mayhem.json`
	}
]
