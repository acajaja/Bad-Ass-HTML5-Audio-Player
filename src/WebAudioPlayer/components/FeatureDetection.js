// FeatureDetection
export const checkWebAudioApiSupport = () => { return Modernizr.audio };
