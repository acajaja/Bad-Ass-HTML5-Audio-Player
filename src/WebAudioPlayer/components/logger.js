/* istanbul ignore file */
const logger = require('pino')({level:'trace'})

export default logger
