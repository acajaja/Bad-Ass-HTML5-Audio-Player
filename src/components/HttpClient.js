// import 'regenerator-runtime/runtime'

export const get = async (endpoint, queryParams = null) => {
    let URL = endpoint;

    if (queryParams) {
        const qp = new URLSearchParams(queryParams);
        URL = `${URL}?${qp.toString}`;
    }

    const requestParams = {
        headers: new Headers({
            'X-Requested-With': ''
        }),
        method: 'GET',
        mode: 'same-origin'
    };

    const response = await fetch(URL, requestParams);

    if (!response.ok) {
        throw new Error('Bad');
    }

    return await response.json();
}
