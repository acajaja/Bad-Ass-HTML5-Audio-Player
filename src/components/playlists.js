export const playlists = [
	{
		"name":"East 3rd Street Ensemble Samples",
		"url":`${process.env.BASE_URL}/playlists/east-third-st.json`
	},
	{
		"name":"Prajna Samples",
		"url":`${process.env.BASE_URL}/playlists/prajna.json`
	},
    {
		"name":"Zuk - An Evening in the Clouds",
		"url":`${process.env.BASE_URL}/playlists/zuk.json`
	},
    {
		"name":"1975: An Underground Epic Samples",
		"url":`${process.env.BASE_URL}/playlists/1975.json`
	},
    {
		"name":"Round Robin Sample",
		"url":`${process.env.BASE_URL}/playlists/round-robin.json`
	},
    {
		"name":"Marim-bass Samples",
		"url":`${process.env.BASE_URL}/playlists/marim-bass.json`
	},
    {
		"name":"Acoustic Mayhem Sample",
		"url":`${process.env.BASE_URL}/playlists/acoustic-mayhem.json`
	}
]
