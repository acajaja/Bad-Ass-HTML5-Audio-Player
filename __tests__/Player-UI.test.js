import * as HttpClient from '../src/WebAudioPlayer/components/HttpClient';
jest.mock('../src/WebAudioPlayer/components/HttpClient.js', () => {
    const mockGet = jest.fn().mockImplementation(() => Promise.resolve(JSON.parse(`{
            "name": "Ttttt",
            "info": "",
            "tracks":
            [
                {
                    "copyright":"2007 East 3rd Street Ensemble",
                    "path":"./audio/short-1.mp3",
                    "info":"",
                    "mime":"audio/mpeg",
                    "title":"Track 1"
                }
            ]
        }`)));

    return {
        __esModule: true,
        get: mockGet
    }
});

import * as WebAudioPlayer from '../src/WebAudioPlayer/WebAudioPlayer';
import { _checkHtml5AudioSupport, __RewireAPI__ as WebAudioPlayerRewireAPI } from '../src/WebAudioPlayer/WebAudioPlayer';

describe('Web Audio Player UI Tests', () => {
    const path      = require('path');
    const { JSDOM } = require('jsdom');

    let audioPage, playaNode;
    let result;

    beforeEach(async () => {
        audioPage = await JSDOM.fromFile(path.resolve(__dirname, 'Stubs/audio-player.html'));
        audioPage.window.HTMLAudioElement.addEventListener = jest.fn((event, callback) => {
            audioPage.window.HTMLAudioElement[event] = jest.fn(callback);
        });
        audioPage.window.HTMLAudioElement['load'] = jest.fn();
        audioPage.window.HTMLAudioElement['play'] = jest.fn(async () => { await 1 + 1; });
        audioPage.window.HTMLAudioElement['seekable'] = {end: jest.fn()};
        WebAudioPlayerRewireAPI.__Rewire__('_canPlayType', () => {
            return 'probably';
        });
        WebAudioPlayerRewireAPI.__Rewire__('_checkWebAudioApiSupport', () => {
            return true;
        });

        playaNode   = audioPage.window.document.getElementById('my-boom-box');
        result      = WebAudioPlayer.init(
            playaNode, audioPage.window.document, audioPage.window.HTMLAudioElement);
    });

    afterEach(() => {
        // restore the original func after test_setUpFunctionality
        jest.resetModules();
    });

    it('Playlists list loaded', () => {
        const playlist = playaNode.querySelector('.playlist-scroll-box');
        const playlistItems = playlist.querySelectorAll('li');
        const playButton = playaNode.querySelector('.play-btn');
        const autoplayButton = playaNode.querySelector('.autoplay-btn');
        const nextButton = playaNode.querySelector('.next-track-btn');
        const prevButton = playaNode.querySelector('.prev-track-btn');
        const infoButton = playaNode.querySelector('.info-button');

        expect(result).toBe(audioPage.window.HTMLAudioElement);
        expect(playlist).toBeInstanceOf(audioPage.window.HTMLOListElement);
        expect(playlistItems.length).toBe(7);
        expect(playlist).toMatchSnapshot(playlist);
        expect(playButton).toBeInstanceOf(audioPage.window.HTMLButtonElement);
        expect(playButton.disabled).toBe(true);
        expect(autoplayButton).toBeInstanceOf(audioPage.window.HTMLButtonElement);
        expect(autoplayButton.disabled).toBe(true);
        expect(nextButton).toBeInstanceOf(audioPage.window.HTMLButtonElement);
        expect(nextButton.disabled).toBe(true);
        expect(prevButton).toBeInstanceOf(audioPage.window.HTMLButtonElement);
        expect(prevButton.disabled).toBe(true);
        expect(infoButton).toBeInstanceOf(audioPage.window.HTMLButtonElement);
        expect(infoButton.disabled).toBe(true);
        expect(playaNode.classList.contains('paused')).toBe(true);
    });

    it('Playlist clicked on', () => {
        const playlistButton = playaNode.querySelector('.playlist-scroll-box li:first-child > button');
        const screenTitle = playaNode.querySelector('.screen-title');
        const scrollBox = playaNode.querySelector('.playlist-scroll-box');
        const mobserver = new audioPage.window.MutationObserver((mutationsList, observer) => {
            for (const mutation of mutationsList) {
                if (mutation.type === 'childList') {
                    expect(mutation.target).toBeInstanceOf(audioPage.window.HTMLParagraphElement);
                    expect(mutation.target.innerHTML).toBe('Ttttt');
                    expect(scrollBox.childElementCount).toBe(1);
                }
            }

            observer.disconnect();
        });
        const config = { childList: true };
    	mobserver.observe(screenTitle, config);

        playlistButton.click();
    });

    it('Playlist loaded, Track clicked on', () => {
        const playlistButton = playaNode.querySelector('.playlist-scroll-box li:first-child > button');
        const scrollBox = playaNode.querySelector('.playlist-scroll-box');

        const mobserver = new audioPage.window.MutationObserver(async (mutationsList, observer) => {
            expect(scrollBox.childElementCount).toBe(1);
            expect(mutationsList[0].type).toBe('childList');

            const li = scrollBox.querySelector('li:first-child');
            const configB = {attributes: true};

            observer.observe(li, configB);

            const trackButton = li.querySelector('button');
            trackButton.click();

            try {
                await audioPage.window.HTMLAudioElement.canplay();
                audioPage.window.HTMLAudioElement.playing();

                expect(li.classList.length).toBe(1);
                expect(li.classList.contains('current')).toBe(true);

                observer.disconnect();
            }
            catch (err) {
                observer.disconnect();
                throw err;
            }
        });

        const config = {childList: true};
    	mobserver.observe(scrollBox, config);
        playlistButton.click();
    });

    it('Playlist loaded, play button clicked on', () => {
        const playButton = playaNode.querySelector('.play-btn');
        const playlistButton = playaNode.querySelector('.playlist-scroll-box li:first-child > button');
        const scrollBox = playaNode.querySelector('.playlist-scroll-box');

        const mobserver = new audioPage.window.MutationObserver(async (mutationsList, observer) => {
            expect(scrollBox.childElementCount).toBe(1);
            expect(mutationsList[0].type).toBe('childList');

            const li = scrollBox.querySelector('li:first-child');
            const configB = {attributes: true};
            observer.observe(li, configB);
            playButton.click();

            try {
                await audioPage.window.HTMLAudioElement.canplay();
                audioPage.window.HTMLAudioElement.playing();

                expect(li.classList.length).toBe(1);
                expect(li.classList.contains('current')).toBe(true);
                expect(playaNode.classList.contains('playing')).toBe(true);

                observer.disconnect();
            }
            catch (err) {
                observer.disconnect();
                throw err;
            }
        });

        const config = {childList: true};
    	mobserver.observe(scrollBox, config);
        playlistButton.click();
    });

    it('Playlist loaded, autoplay button clicked on', () => {
        const autoplayButton = playaNode.querySelector('.autoplay-btn');

        expect(autoplayButton.disabled).toBe(true);
        const playlistButton = playaNode.querySelector('.playlist-scroll-box li:first-child > button');
        const scrollBox = playaNode.querySelector('.playlist-scroll-box');

        const mobserver = new audioPage.window.MutationObserver(async (mutationsList, observer) => {
            expect(scrollBox.childElementCount).toBe(1);

            for (const m of mutationsList) {
                if (m.type === 'childList') {
                    autoplayButton.click();

                    if (playaNode.classList.length === 4) {
                        expect(playaNode.classList.contains('autoplay')).toBe(true);
                    }
                }
            };

            mobserver.disconnect();
        });

        const config = {childList: true};
    	mobserver.observe(scrollBox, config);
        playlistButton.click();
    });

    it('Playlists loaded, volume button clicked on', () => {
        const volumeButton = playaNode.querySelector('.volume-button');
        const volumeSliderBox = playaNode.querySelector('.volume-slider-mute-box');

        const mobserver = new audioPage.window.MutationObserver(async (mutationsList, observer) => {
            for (const m of mutationsList) {
                if (m.type === 'attributes') {
                    expect(m.target.classList.contains('play')).toBe(true);
                    expect(m.target.querySelector('input').disabled).toBe(true);
                    expect(m.target.querySelector('button').disabled).toBe(true);
                }
            };

            mobserver.disconnect();
        });

        const config = {attributes: true};
    	mobserver.observe(volumeSliderBox, config);
        volumeButton.click();
    });

    it('Playlist loaded, volume button clicked on', () => {
        const volumeButton = playaNode.querySelector('.volume-button');
        const volumeSliderBox = playaNode.querySelector('.volume-slider-mute-box');
        const playlistButton = playaNode.querySelector('.playlist-scroll-box li:first-child > button');
        const scrollBox = playaNode.querySelector('.playlist-scroll-box');

        const mobserver = new audioPage.window.MutationObserver(async (mutationsList, observer) => {
            expect(scrollBox.childElementCount).toBe(1);

            for (const m of mutationsList) {
                if (m.type === 'childList') {
                    const config = {attributes: true};
                	mobserver.observe(volumeSliderBox, config);
                    volumeButton.click();

                    console.debug(m.type);
                    if (m.type === 'attributes') {
                        expect(volumeSliderBox.classList.contains('play')).toBe(true);
                        expect(m.target.querySelector('input').disabled).toBe(false);
                        expect(m.target.querySelector('button').disabled).toBe(false);
                    }
                }
            };

            mobserver.disconnect();
        });

        const config = {childList: true};
    	mobserver.observe(scrollBox, config);
        playlistButton.click();
    });
});
