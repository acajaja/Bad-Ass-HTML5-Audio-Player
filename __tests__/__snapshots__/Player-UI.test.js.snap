// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Web Audio Player UI Tests Playlists list loaded 1`] = `Object {}`;
